language_service/src/definition.rs

```diff
@@ -59,6 +59,28 @@ impl<'a> Handler<'a> for DefinitionFinder<'a> {
         ));
     }
 
+    fn at_type_param_def(
+        &mut self,
+        _: &LocatorContext<'a>,
+        def_name: &'a ast::Ident,
+        _: hir::ty::ParamId,
+    ) {
+        self.definition = Some(protocol_location(self.compilation, def_name.span, None));
+    }
+
+    fn at_type_param_ref(
+        &mut self,
+        context: &LocatorContext<'a>,
+        _: &'a ast::Ident,
+        param_id: hir::ty::ParamId,
+    ) {
+        if let Some(curr) = context.current_callable {
+            if let Some(def_name) = curr.generics.get(usize::from(param_id)) {
+                self.definition = Some(protocol_location(self.compilation, def_name.span, None));
+            }
+        }
```

Comment:
Symmetry: Somehow it bothers me that the `at_local_ref` and `at_type_param_ref` signatures (and their go-to-def implementations) don't look more alike. They're sooo similar in how they're treated that I think the method parameters should also map exactly.

We have:

```rs
fn at_local_ref(
    &mut self,
    context: &LocatorContext<'package>, // context
    path: &'package ast::Path, // reference node
    node_id: &'package ast::NodeId, // id (for finding references etc)
    ident: &'package ast::Ident, // declaration node
);
```

We have
```rs
fn at_type_param_ref(
    &mut self,
    context: &LocatorContext<'a>, // context
    _: &'a ast::Ident, // reference node
    param_id: hir::ty::ParamId, // id (for finding references etc)
// no declaration node parameter
);
```

_Almost_ the same. All it's missing is the declaration node at the end. You're looking that up here in `definition.rs`. But for locals, you do the equivalent lookup in `name_locator.rs`. I feel like either they should _both_ pass in the `ast::Ident` for the declaration, or they should _both_ omit it and let `definition.rs` look it up.

Reply:
I've opted to add the parameter to the `at_type_para_ref` signature as you suggested.